pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -323,6 +322,92 @@ public enum Type {
       public static final String CONFIG_OF_MAX_RETRIES = "pinot.broker.failure.detector.max.retries";
       public static final int DEFAULT_MAX_RETIRES = 10;
     }
+
+    // Configs related to AdaptiveServerSelection.
+    public static class AdaptiveServerSelector {
+      /**
+       * Adaptive Server Selection feature has 2 parts:
+       * 1. Stats Collection
+       * 2. Routing Strategy
+       *
+       * Stats Collection is controlled by the config CONFIG_OF_ENABLE_STATS_COLLECTION.
+       * Routing Strategy is controlled by the config CONFIG_OF_TYPE.
+       *
+       *
+       * Stats Collection: Enabling/Disabling stats collection will dictate whether stats (like latency, #
+       *                   inflight requests) will be collected when queries are routed to/received from servers.
+       *                   It does not have any impact on the Server Selection Strategy used.
+       *
+       * Routing Strategy: Decides what strategy should be used to pick a server. The available strategies are as
+       *                   follows:
+       *                   1. NO_OP: Uses the default roundrobin approach. Does NOT require Stats Collection to be
+       *                   enabled.
+       *                   2. NUM_INFLIGHT_REQ: Picks the best server based on the number of inflight requests for
+       *                   each server. Requires Stats Collection to be enabled.
+       *                   3. LATENCY: Picks the best server based on the Exponential Weighted Moving Averge of Latency
+       *                   for each server. Requires Stats Collection to be enabled.
+       *                   4. HYBRID: Picks the best server by computing a custom hybrid score based on both latency
+       *                   and # inflight requests. Requires Stats Collection to be enabled.
+       */
+
+      public enum Type {
+        // Adaptive Selectors are not used. Use default round-robin.
+        NO_OP,
+
+        // Use selector that routes to servers based on number of inflight requests.
+        NUM_INFLIGHT_REQ,
+
+        // Use selector that routes to servers based on observed latencies.
+        LATENCY,
+
+        // Uses both latency and numInFlightRequests to select the best server based on C3 score. It is based on the
+        // approach described in https://www.usenix.org/system/files/conference/nsdi15/nsdi15-paper-suresh.pdf
+        HYBRID
+      }
+
+      // Determines the type of AdaptiveServerSelector to use.
```

Comment:
(nit) suggest adding a brief comment for future readers to be aware of the decision and reasoning behind using 2 configs.

Reply:
Done.